bolt-modules/boltlib/lib/puppet/functions/run_task.rb

```diff
@@ -89,7 +89,8 @@ def run_task_with_description(task_name, targets, description, args = {})
       task = Bolt::Task.from_task_signature(task_signature)
 
       # Set the default value for any params that have one and were not provided
-      params = task.parameter_defaults.merge(params)
+      # If a parameter was given an undef value and has a default specified, use the default
+      params = task.parameter_defaults.merge(params) { |_, default, passed| passed || default }
```

Comment:
This will cause a value of `false` to also be overridden by the default. I think we should just strip out any params with value `nil` at the start.